2.0/ingest/text_embedding.py

```
@@ -0,0 +1,92 @@
+        trans = conn.begin()
+        
+        try:
+            # Get unembedded sections
+            result = conn.execute(text(f'''
+                SELECT id, {text_column} 
+                FROM {source_table} 
+                WHERE id NOT IN (
+                    SELECT section_id FROM {embeddings_table}
+                )
+                ORDER BY id
+            '''))
```

Ruff (0.13.3)

Possible SQL injection vector through string-based query construction

(S608)